### PR TITLE
Search engine

### PR DESCRIPTION
**Pull request by aleks279**
Search engine functionality for:
- Events, with filters: `event type & name`
- Bands, with filters: `genre, event type & name`
- Musicians, with filters: `genre, event type & name`

Go nuts.

FYI: this also happens to fix a little mishap in the `schema.rb`

### CODE REVIEW

**Comment by 4RN4LDO** [comment]
Looks good to me 🌵

File: app/views/partials/_cover_search.html.haml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 .cover
   .search-bar
     %i.icon-search
-    %input#search-bar-text{placeholder: "search something", type: "text", value: ""}
+    -#%input#search-bar-text{placeholder: "search something", type: "text", value: ""}
```
Comment (jramoscr) [inline app/views/partials/_cover_search.html.haml:4]: Why this line is commented? If it's not necessary, remove it.

File: app/models/event.rb [hunk 2]
```diff
@@ -1,4 +1,9 @@
 class Event < ActiveRecord::Base
+  include Filterable
+
+  scope :event_type, -> (event_type) { where event_type_id: event_type}
```
Comment (jramoscr) [inline app/models/event.rb]: Use this with the parenthesis to keep consistency: `where(event_type_id: event_type)`; also; are you passing an id or the entire object to this scope? Maybe here we can change the param name to `event_type_id`, in this way we know what we're passing explicitly. What do you think?

File: app/views/partials/_cover_search.html.haml (hunk 1 above)
Comment (aleks279) [inline app/views/partials/_cover_search.html.haml:4]: Problem is, I don't know yet if it's gonna be necessary. I'll update the file in question after tomorrow's meeting with the client.

**Pull request merged**